src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala

```diff
@@ -421,4 +417,18 @@ class MarathonSchedulerService @Inject() (
       reconciliationInitialDelay.toMillis + reconciliationInterval.toMillis
     )
   }
+
+  private def startAbdication(abdicationCommand: () => Unit, driverHandlesAbdication: Boolean): Unit = {
```

Comment:
We should find a better name for this method and also the parameter `driverHandlesAbdication`. I would even question the utility of this flag. I know that you are preserving existing functionality but I think the functionality does not make sense. The flag is only set to `false` if `resetOfferLeadershipBackOff` throws an exception. Not very likely.